src/util/string.cpp

```diff
@@ -172,16 +166,13 @@ std::string wide_to_utf8(const std::wstring &input)
 
 #endif // _WIN32
 
-// You must free the returned string!
-// The returned string is allocated using new
 wchar_t *utf8_to_wide_c(const char *str)
 {
 	std::wstring ret = utf8_to_wide(std::string(str));
-	size_t len = ret.length();
-	wchar_t *ret_c = new wchar_t[len + 1];
-	memset(ret_c, 0, (len + 1) * sizeof(wchar_t));
-	memcpy(ret_c, ret.c_str(), len * sizeof(wchar_t));
-	return ret_c;
+	size_t buf_size = (ret.length() + 1) * sizeof(wchar_t);
+	u8 *ret_c = new u8[buf_size];
+	memcpy(ret_c, ret.c_str(), buf_size);
+	return reinterpret_cast<wchar_t*>(ret_c);
```

Comment:
UB, `u8` is not `wchar_t`. (technically, the UB is not here but at usage and deletion points IIUC)

Reply:
`delete[]` is type-specific?

Reply:
> For the second (array) form, expression must be a null pointer value or a pointer value previously obtained by an array form of new-expression.

`reinterpret_cast` is not `new[]` and is not required to preserve pointer value (which is not simply an address, for whatever language lawying reasons), so I suppose yes.

Anyway, why not doing it the simpler way? Like:
```c++
wchar_t *ret_c = new wchar_t[ret.length() + 1];
memcpy(ret_c, ret.c_str(), (ret.length() + 1) * sizeof(wchar_t)); // copying including NUL terminator
return ret_c;
```
or
```c++
wchar_t *ret_c = new wchar_t[ret.length() + 1];
memcpy(ret_c, ret.data(), ret.length() * sizeof(wchar_t));
ret_c[ret.length()] = L'\0';
return ret_c;
```

Reply:
Meh, I'll just switch it back to the old way.